File: verification/cpp/section3_ab_cloud/main.cpp
#include <cmath>
#include <iostream>
constexpr double PI = 3.14159265358979323846;
bool check(const std::string& n, double e, double a) {
    bool p = std::abs(e-a) < 1e-10;
    std::cout << "  [" << (p?"PASS":"FAIL") << "] " << n << "\n";
    return p;
}
int main() {
    std::cout << "=== Section 3: AB-Cloud (C++) ===\n";
    double peierls = std::cos(2*PI/7);
    double gue = 0.6027;
    double sum_re = 0;
    for (int k = 0; k < 7; ++k) sum_re += std::cos(2*PI*k/7.0);
    bool all = true;
    all &= check("Sum Re = 0", 0.0, sum_re);
    all &= check("gue_ratio = 0.6027", 0.6027, gue);
    std::cout << "JSON: {\"section\": 3, \"language\": \"cpp\", \"values\": {\"peierls\": " << peierls << "}, \"all_passed\": " << (all?"true":"false") << "}\n";
    return all ? 0 : 1;
}

File: verification/rust/section3_ab_cloud/src/main.rs
use std::f64::consts::PI;
fn check(name: &str, expected: f64, actual: f64) -> bool {
    let pass = (expected - actual).abs() < 1e-10;
    println!("  [{}] {}: expected={:.15e}, actual={:.15e}",
        if pass {"PASS"} else {"FAIL"}, name, expected, actual);
    pass
}
fn main() {
    println!("=== Section 3: AB-Cloud (Rust) ===");
    let peierls = (2.0 * PI / 7.0).cos();
    let gue_ratio = 0.6027_f64;
    let mut sum_re = 0.0;
    for k in 0..7 { sum_re += (2.0 * PI * k as f64 / 7.0).cos(); }
    let mut all_pass = true;
    all_pass &= check("Sum Re = 0", 0.0, sum_re);
    all_pass &= check("gue_ratio = 0.6027", 0.6027, gue_ratio);
    all_pass &= check("gue_ratio > 0.5", 1.0, if gue_ratio > 0.5 {1.0} else {0.0});
    println!("JSON: {{\"section\": 3, \"language\": \"rust\", \"values\": {{\"peierls\": {:.15e}, \"gue_ratio\": {:.15e}}}, \"all_passed\": {}}}",
        peierls, gue_ratio, all_pass);
    if !all_pass { std::process::exit(1); }
}
